app/controllers/api/v1/notifications_controller.rb

```diff
@@ -9,8 +9,12 @@ class Api::V1::NotificationsController < Api::BaseController
   DEFAULT_NOTIFICATIONS_LIMIT = 40
 
   def index
-    @notifications = load_notifications
-    render json: @notifications, each_serializer: REST::NotificationSerializer, relationships: StatusRelationshipsPresenter.new(target_statuses_from_notifications, current_user&.account_id)
+    ApplicationRecord.connected_to(role: :read, prevent_writes: true) do
```

Comment:
If you keep adding these to more and more places, a helper like `on_read_replica` or `with_read_only` or something which wraps up the `ApplicationRecord.connected_to(role: :read, prevent_writes: true)` call might be nice DRY/syntax helper to add somewhere.

Reply:
Indeed